CoreFoundation/URL.subproj/CFURL.c

```diff
@@ -4534,13 +4534,12 @@ CFStringRef CFURLCreateStringWithFileSystemPath(CFAllocatorRef allocator, CFURLR
 }
 
 Boolean CFURLGetFileSystemRepresentation(CFURLRef url, Boolean resolveAgainstBase, uint8_t *buffer, CFIndex bufLen) {
-#if TARGET_OS_MAC || TARGET_OS_LINUX || TARGET_OS_WIN32
     CFAllocatorRef alloc = CFGetAllocator(url);
     CFStringRef path;
 
     if (!url) return false;
-#endif
-#if TARGET_OS_MAC || TARGET_OS_LINUX
+
+#if TARGET_OS_MAC || TARGET_OS_LINUX || TARGET_OS_BSD
```

Comment:
Would it be better to make this an `#if !TARGET_OS_WIN32` else this function will always return `false` when run on something that isn't a listed platform?

Reply:
CF only mentions `kCFURLPOSIXPathStyle` or `kCFURLWindowsPathStyle`(`kCFURLHFSPathStyle` is deprecated). So I think in this case your suggestion makes sense to make this a binary conditional.

However, we should probably invert sense in that case: TARGET_OS_WIN32 comes first, and POSIX is an `#else`.